cylc/flow/task_job_mgr.py

```diff
@@ -1345,6 +1348,9 @@ def get_job_conf(
             'pre-script': rtconfig['pre-script'],
             'script': rtconfig['script'],
             'submit_num': itask.submit_num,
+            'symlink_dirs': get_dirs_to_symlink(
```

Comment:
If this were to go forward, I imagine making sure `get_dirs_to_symlink` was cached would be a reasonable thing to do for a minor speedup with limited extra memory load.